ovs/lib/scheduledtask.py

```diff
@@ -277,13 +282,50 @@ def gather_scrub_work():
                                                                               vdisk_guids=local_vdisks_to_scrub)
             except Exception as ex:
                 logger.error('Gather Scrub - Storage Router {0:<15} - Scrubbing failed with error:\n - {1}'.format(local_storage_router.ip, ex))
-        all_results = result_set.join(propagate=False)  # Propagate False makes sure all jobs are waited for even when 1 or more jobs fail
+
+        all_results, failed_nodes = manage_running_tasks(result_set, timesleep=60)  # Check every 60 seconds if tasks are still running
+
+        for index, result in enumerate(all_results):
+            if isinstance(result, list):
+                processed_guids.extend(result)
+            else:
+                logger.error('Gather Scrub - Storage Router {0:<15} - Scrubbing failed with error:\n - {1}'.format(storage_router_list[index].ip, result))
```

Comment:
Are we sure the order of ```storage_router_list``` matches the order in which the ```manage_running_tasks``` returns the results for each storagerouter?? Because ```result_set``` which is passed to the function is a dict, which is obviously unordered